Documentation/design-docs/framework-version-resolution-implementation.md

```diff
@@ -0,0 +1,43 @@
+# Breaking changes
```

Comment:
If this is going into `hostfxr` then do we need to support the old algorithm for older apps? 
Are we intentionally changing 2.x behavior?
Do we need shiproom approval? Or does this fall into the category of other muxer/hostfxr changes.  Must be done by...

Reply:
This is the money question. I personally think we're OK because:
* Behavior on release versions is almost 100% backward compatible (the minor changes are in general relaxing, "used to fail" -> "now it works").
* Behavior on pre-release is not that interesting, since by now there should be very few 2.* apps which refer to pre-release versions. And the changes are not "hard breaking" as in something which worked now fails. It's more about we picked some version before, now we may pick a different version. Which may be breaking for some apps, but it's much less likely.

So to answer:
- Yes, we're intentionally changing 2.* behavior (in fact any version behavior)
- Yes, some of these changes are technically breaking
- No, I don't think we need to keep the old algorithms around (see the comment above)
- Maybe, we should seek shiproom approval. What I do plan is to compile some reasonably short list of the changes and send it out to interested parties for "Approval" - working on that.

Reply:
I've sent out a report with all the impactful changes... seems like we're going to run it through tactics as well.